modules/phpfpm/decode.go

```diff
@@ -0,0 +1,125 @@
+package phpfpm
+
+import (
+	"bufio"
+	"encoding/json"
+	"errors"
+	"io"
+	"strconv"
+	"strings"
+)
+
+type decoder func(r io.Reader, s *status) error
+
+func decodeJSON(r io.Reader, s *status) error {
+	return json.NewDecoder(r).Decode(s)
+}
+
+func decodeText(r io.Reader, s *status) error {
```

Comment:
overall decodeText func is hard to follow, how about to split it into several smaller ones?